main.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+def verify_input(array)
+  if !array.is_a?(Array)
+    false
+  else
+    array.respond_to?('each') ? true : false
+  end
+end
+
+def bubble_sort(arr)
+  return unless verify_input(arr)
+
+  length = arr.length
+  loop do
+    sorted = true
+    (length - 1).times do |i|
+      next if i + 1 == arr.length
+
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        sorted = false
+      end
+    end
+    break if sorted
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  if verify_input(arr)
+    if block_given?
+      loop do
```

Comment:
check your block, the error states: expected Exception but nothing was raised

        bubble_sort_by raises an error if block is not given
        Failure/Error: expect { bubble_sort_by(unsorted) }.to raise_error expected Exception but nothing was raised

Reply:
Oh, ok, the problem is that we can't find the tests that you are referring. Can you tell us where are such tests please? Also there was nothing about exceptions on the requirements, but we can add that exception without problem. Thanks in advance!